### PR TITLE
Integration tests using Testcontainers

### PR DESCRIPTION
**Pull request by alingrig87**
- create integration tests for GreetingsService and GreetingsController
image

- use Testcontainers for using a different MongoDB 
image

### CODE REVIEW

File: Could
Comment (vgrigoriu) [inline Could:1]: Do we need this file? And `Task` below?

**Review by vgrigoriu — APPROVED** [review]

File: Could
Comment (alingrig87) [reply Could:1]: addressed in 0c987922819ee511b96d6246f99a32d117b2a799

File: src/test/java/ro/unibuc/hello/controller/GreetingsControllerIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        mockMvc.perform(delete("/greetings/1"))
+            .andExpect(status().isOk());
+
+        mockMvc.perform(get("/greetings"))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.length()").value(1))
+                .andExpect(jsonPath("$[0].content").value("Hello 2"));
+    }
+}
```
Comment (alexOporanu1) [inline src/test/java/ro/unibuc/hello/controller/GreetingsControllerIntegrationTest.java:139]: Nit: no new line at end of file, there are a few more files with this issue

Comment (alingrig87) [reply src/test/java/ro/unibuc/hello/controller/GreetingsControllerIntegrationTest.java]: @alexOporanu1, addressed in 653d7df000551cd920d156103c5c4fc9760f5d84, thank you.

**Review by alexOporanu1 — APPROVED** [review]

**Pull request merged**